tests/lib/sonar.ts

```diff
@@ -198,6 +198,143 @@ test.serial(`If config.rules has some rules "off", we shouldn't create those rul
     t.context.eventemitter.prototype.on.restore();
 });
 
+test.serial('If config.rules is an array with rules, we should create just those rules', (t) => {
+    const rule = {
+        create() {
+            return {};
+        },
+        meta: {}
+    };
+
+    sinon.spy(eventEmitter.EventEmitter2.prototype, 'on');
+    t.context.rule = rule;
+    sinon.stub(t.context.resourceLoader, 'loadRules').returns(new Map([
+        ['disallowed-headers', rule],
+        ['lang-attribute', rule],
+        ['manifest-exists', rule]
+    ]));
+    sinon.stub(rule, 'create')
+        .onFirstCall()
+        .returns({ 'fetch::end': () => { } })
+        .onSecondCall()
+        .returns({ 'fetch::error': () => { } });
+
+    const sonarObject = new Sonar({ //eslint-disable-line no-unused-vars
+        collector: 'collector',
+        rules: [
+            'disallowed-headers:warning',
+            'manifest-exists:warning'
+        ]
+    });
+
+    t.true(t.context.resourceLoader.loadRules.called);
+    t.true(t.context.rule.create.calledTwice);
+    t.true(t.context.eventemitter.prototype.on.calledTwice);
```

Comment:
I think we need to update this test and the previous one. `loadRules` should just return 2 rules now that we only load the ones we need. My mistake for not updating it in my PR :(